### ISSUE TITLE
OAuthLib FAQ 

### ISSUE THREAD

**Issue by ib-lundgren** [oauthlib/oauthlib#119]
OAuthLib is steadily covering more and more of the OAuth specs and consequently it can simplify a number of OAuth related problems for users. There needs to be a short and sweet collection of common problems and their solutions. Short solutions could simply be inlined and longer linked to...
- Do you support XYZ? See #117
- What is an OAuth client?
- (Client) Connecting to various common services like Google, Facebook, Twitter
- OAuth 1) What is a client_key, client_secret, etc
- OAuth 2) What is a client_id, scope, state, etc
- What is the difference between authentication and authorization?
- How do I secure my users and API using OAuth?
- What is an OAuth provider?
- In a nutshell, what is the difference between OAuth 1 & 2?
- Some argue OAuth 2 is worse than 1, is that true?
- When to use OAuth 1 and 2 (#118).
- How to get started on an OAuth 1 provider
- How to get started on an OAuth 2 provider
  ...

Please add more suggestions =)
